prow/lib.sh

```diff
@@ -41,12 +41,14 @@ function trace() {
 }
 
 function setup_gcloud_credentials() {
-  if [[ $(command -v gcloud) ]]; then
-    gcloud auth configure-docker -q
-  elif [[ $(command -v docker-credential-gcr) ]]; then
-    docker-credential-gcr configure-docker
-  else
-    echo "No credential helpers found, push to docker may not function properly"
+  if [[ -n "${GOOGLE_APPLICATION_CREDENTIALS:-}" ]]; then
```

Comment:
We should not check for `GOOGLE_APPLICATION_CREDENTIALS` ideally, this is the legacy mode where we mount SA token. Jobs are now using workload identity.

I agree we need some check but I am not sure it should be this. Unfortunately I don't know a great way to actually identity that. If we just add a `|| true` is it sufficient or do we need something more?

Reply:
Yes, actually we need a way to bypass the setup the gcloud credentials for docker if it does not exist. We just choosed this way to bypass the adding credHelpers to docker in our internal CI.

I see the env GOOGLE_APPLICATION_CREDENTIALS value is still in istio/test-infra/prow/cluster/jobs/all-presets.yaml, so does the "workload identity" here mean?

Reply:
It's defined there but no jobs use that anymore https://cloud.google.com/kubernetes-engine/docs/how-to/workload-identity